plugins/Actions/DataTable/Filter/Actions.php

```diff
@@ -30,11 +32,27 @@ public function __construct($table)
     public function filter($table)
     {
         $table->filter(function (DataTable $dataTable) {
+
+            $defaultActionName = Config::getInstance()->General['action_default_name'];
+
             foreach ($dataTable->getRows() as $row) {
                 $url = $row->getMetadata('url');
                 if ($url) {
                     $row->setMetadata('segmentValue', urldecode($url));
                 }
+
+                // remove the default action name 'index' in the end of flattened urls
+                if (Common::getRequestVar('flat', 0)) {
+                    $label = $row->getColumn('label');
+                    if (substr($label, -strlen($defaultActionName)) == $defaultActionName) {
+                        $label = substr($label, 0, -strlen($defaultActionName));
+                        if (substr($label, -1) == '/') {
+                            $label = rtrim($label, '/') . '/';
+                        }
```

Comment:
I think you can just have `$label = rtrim($label, '/') . '/';` here instead of the `if()`, right? It should have the same effect even if the string doesn't end w/ '/'.

Reply:
Also does this code need to take into account the `[General] action_url_category_delimiter` option? I don't know what the use case would be for it not to be `/`, but you never know I guess.

Reply:
using the config should make sense